### ISSUE TITLE
Tags should not be required to be a keyword list, and should support nesting

### ISSUE THREAD

**Issue by GregMefford** [spandex-project/spandex#92]
Ultimately, these get encoded back into a map and sent to the collector, so there's no reason to enforce that they be a Keyword list (i.e. atom keys).

For example, it would be nice to be able to include some map of key-value pairs that were decoded from a JSON payload, but you don't want to convert all the keys to atoms. Also in this scenario, it would be nice to be able to have nested maps as values. On a Datadog-specific note: I believe this is possible, but we'd have to traverse the maps and change the names to a flat structure of dot-syntax keys like `parent.child.key: value`.

**Comment by zachdaniel** [comment]
Yeah, that makes sense to me. We should be able to change the type of that option to `[:map, :keyword]`, and then add code to handle that change on the sender side. I think we also want to make sure that they all get merged correctly.

I'm pretty sure merging tags is the right answer, at least. I'm pretty confident that it is.

**Comment by lswith** [comment]
Here is an example of how tags can be used:
` tags = %{ "elasticsearch.body" => body, "elasticsearch.method" => "GET", "elasticsearch.endpoint" => endpoint }`

This then allows metadata to each span to something like this:
![monitor-elasticsearch-datadog-apm-error-stack-trace-v3](https://user-images.githubusercontent.com/1018619/53769843-2f3d3a80-3f31-11e9-8a48-110121ec8221.png)

(from this article here: https://www.datadoghq.com/blog/monitor-elasticsearch-datadog/)

**Comment by eraserhd** [comment]
It's possible to use keywords with periods in them, using a slightly different elixir syntax, for example, we have this:

```elixir
config :db, Db.Datadog.Tracer,
  tags: [
    {:"git.commit.sha", git_commit_sha},
    {:"git.repository_url", git_repo_url}
  ]
```

But you can even do this:

```elixir
config :db, Db.Datadog.Tracer,
  tags: [
    "git.commit.sha": git_commit_sha,
    "git.repository_url": git_repo_url
  ]
```

(Those are actually keywords and not string keys, it's using a quoted atom syntax.)
